zeromq/pyzmq
# Update appveyor testing

This PR updates the current appveyor test setup to run all the tests on Windows

in more details:

- Skip tests which are not supported on windows (e.g. using file handles for sockets).
- Skip `TestContext.test_destroy` because it segfaults
- Fix Skip behavior in `TestEINTRSysCall`.  `Pytest` cannot detect items marked as skip if they are not test methods.
- Add/increase timeouts or time.sleep() where necessary to consider slow VM builds.
- Run the `test_win32_shim` test on a separate/new console to avoid issues with the CTRL-C event been sent to all the processes which share the console and abort the appveyor build prematurely.
- Cache wheels (minor build speedups)

The builds are still a little flaky and will need to be rerun from time to time. 
This PR should address #973
